src/_stories/NewUserModal.stories.js

```diff
@@ -0,0 +1,84 @@
+import { action } from '@storybook/addon-actions'
+import { boolean, number, text, withKnobs } from '@storybook/addon-knobs'
+import { storiesOf } from '@storybook/react'
+import { Fragment, useEffect, useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { NewUserModal } from 'src/components/group-common'
+import { ajaxOverridesStore } from 'src/libs/state'
+
+
+const INVITE_SUCCEEDED = 201
+const USER_NOT_REGISTERED = 404
+const USER_IS_REGISTERED = 200
+const FAILED_REQUEST = 400
+
+const delayed = ({ ms, resolved, status }) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (resolved) {
+        resolve(new Response({ status }))
+      } else {
+        reject({ status, json: () => ({ message: 'Failed to add user' }) })
+      }
+    }, ms)
+  })
+}
+
+const clearOverrides = () => ajaxOverridesStore.set([])
+
+const setOverrides = ({ isRegistered, ms }) => {
+  ajaxOverridesStore.set([{
+    filter: /api\/users\/v1\/(?!invite)/,
+    fn: () => {
+      action('Checking if user is registered')()
+      const status = isRegistered ? USER_IS_REGISTERED : USER_NOT_REGISTERED
+      return delayed({ ms, resolved: isRegistered, status })
+    }
+  }, {
+    filter: /users\/v1\/invite/,
+    fn: () => {
+      action('Requesting user invite')()
+      return delayed({ ms, resolved: true, status: INVITE_SUCCEEDED })
+    }
+  }])
+}
+
+const Modal = () => {
+  const ms = number('Latency', 100)
+  const addUnregisteredUser = boolean('Allow adding unregistered user', true)
+  const isRegistered = boolean('User is registered', false)
+  const addSucceeds = boolean('User added successfully', true)
+  const title = text('Title', 'Add user to Terra Group')
+  const adminLabel = text('Admin Label', 'admin')
+  const userLabel = text('User Label', 'user')
+  const [showModal, setShowModal] = useState(true)
+
+  useEffect(() => clearOverrides, [])
+  setOverrides({ isRegistered, ms })
```

Comment:
TOL: Even though this isn't production code, if we're going to create and maintain it as an artifact we should strive for the same standards of quality. Instead of mutating state in the render path, this should use an effect:
```
useEffect(() => {
  setOverrides({ isRegistered, ms })
  return clearOverrides
}, [isRegistered, ms])
```